syntax = "proto3";

package data_node;

service DataNode {
    rpc Put(PutRequest) returns (PutResponse);
    rpc Get(GetRequest) returns (GetResponse);
    rpc Heartbeat(HeartbeatRequest) returns (HeartbeatResponse);
}

message PutRequest {
    string key = 1;
    string value = 2;
    string keyspace = 3;
}

message PutResponse {
    string status = 1;
}

message GetRequest {
    string key = 1;
    string keyspace = 2;
}

message GetResponse {
    string status = 1;
    string value = 2;
}

message HeartbeatRequest {}

message HeartbeatResponse {
  string status = 1;
}
